backend/man/discord-gating-bot.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH discord-gating-bot 1  "discord-gating-bot 0.1.0" 
.SH NAME
discord\-gating\-bot \- A discord bot that can grant discord roles on a server based on reputation in a specific colony
.SH SYNOPSIS
\fBdiscord\-gating\-bot\fR [\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fB\-c\fR|\fB\-\-config\-file\fR] [\fB\-v\fR|\fB\-\-verbose\fR] [\fB\-q\fR|\fB\-\-quiet\fR] [\fB\-t\fR|\fB\-\-token\fR] [\fB\-i\fR|\fB\-\-invite\-url\fR] [\fB\-h\fR|\fB\-\-host\fR] [\fB\-u\fR|\fB\-\-url\fR] [\fB\-p\fR|\fB\-\-port\fR] [\fB\-d\fR|\fB\-\-directory\fR] [\fB\-s\fR|\fB\-\-storage\-type\fR] [\fB\-k\fR|\fB\-\-key\fR] [\fIsubcommands\fR]
.SH DESCRIPTION
.PP
Running the bot without any sub command will start an http server,
connect to discord and listen for commands, all with the default configuration.
.PP
Running the bot for the first time, you probably want to generate an encryption
key and register the discord slash commands with the `storage` and `discord`
subcommands.
.PP
.PP
.SH OPTIONS
.TP
\fB\-\-help\fR
Print help information
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version information
.TP
\fB\-c\fR, \fB\-\-config\-file\fR=\fIFILE\fR
Sets a custom config file
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Define the verbosity of the application, repeat for more verbosity
.TP
\fB\-q\fR, \fB\-\-quiet\fR
Supress all logging
.TP
\fB\-t\fR, \fB\-\-token\fR=\fITOKEN\fR
The discord bot token
.TP
\fB\-i\fR, \fB\-\-invite\-url\fR=\fIINVITE_URL\fR
The discor bot invitation url
.TP
\fB\-h\fR, \fB\-\-host\fR=\fIHOST\fR
The address to listen on
.TP
\fB\-u\fR, \fB\-\-url\fR=\fIURL\fR
The base url under which the server is reachable
.TP
\fB\-p\fR, \fB\-\-port\fR=\fIPORT\fR
The port to listen on
.TP
\fB\-d\fR, \fB\-\-directory\fR=\fIDIRECTORY\fR
The path where the persistent data is stored
.TP
\fB\-s\fR, \fB\-\-storage\-type\fR=\fISTORAGE_TYPE\fR
How to store data, on disk or in memory
.TP
\fB\-k\fR, \fB\-\-key\fR=\fIKEY\fR
The encryption_key used to encrypt the stored data
.SH SUBCOMMANDS
.TP
discord\-gating\-bot\-config(1)
Print the configuration and get a template file
.TP
discord\-gating\-bot\-storage(1)
Interact with the presistent storage and encryption
.TP
discord\-gating\-bot\-discord(1)
Interact with discord directly, e.g. register slash commands
.TP
discord\-gating\-bot\-help(1)
Print this message or the help of the given subcommand(s)
.SH VERSION
v0.1.0
.SH AUTHORS
Hans Mündelein <hans@muendelein.me>
Christian Maniewski <code@chmanie.com>


backend/completion/discord-gating-bot.bash
_discord-gating-bot() {
    local i cur prev opts cmds
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${i}" in
            "$1")
                cmd="discord__gating__bot"
                ;;
            add)
                cmd+="__add"
                ;;
            config)
                cmd+="__config"
                ;;
            delete)
                cmd+="__delete"
                ;;
            discord)
                cmd+="__discord"
                ;;
            gate)
                cmd+="__gate"
                ;;
            generate)
                cmd+="__generate"
                ;;
            global)
                cmd+="__global"
                ;;
            guild)
                cmd+="__guild"
                ;;
            help)
                cmd+="__help"
                ;;
            list)
                cmd+="__list"
                ;;
            register)
                cmd+="__register"
                ;;
            remove)
                cmd+="__remove"
                ;;
            show)
                cmd+="__show"
                ;;
            storage)
                cmd+="__storage"
                ;;
            template)
                cmd+="__template"
                ;;
            user)
                cmd+="__user"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        discord__gating__bot)
            opts="-V -c -v -q -t -i -h -u -p -d -s -k --help --version --config-file --verbose --quiet --token --invite-url --host --url --port --directory --storage-type --key config storage discord help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --config-file)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -c)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --token)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -t)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --invite-url)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -i)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --host)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -h)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --url)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -u)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --port)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -p)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --directory)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -d)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --storage-type)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --key)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -k)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__config)
            opts="-h -v -q --help --verbose --quiet show template help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__config__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__config__show)
            opts="-h -v -q --help --verbose --quiet"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__config__template)
            opts="-h -v -q --help --verbose --quiet"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord)
            opts="-h -v -q --help --verbose --quiet register delete help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__delete)
            opts="-h -v -q --help --verbose --quiet global guild help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__delete__global)
            opts="-h -v -q --help --verbose --quiet"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__delete__guild)
            opts="-h -v -q --help --verbose --quiet <GUILD_ID>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__delete__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__register)
            opts="-h -v -q --help --verbose --quiet global guild help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__register__global)
            opts="-h -v -q --help --verbose --quiet"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__register__guild)
            opts="-h -v -q --help --verbose --quiet <GUILD_ID>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__discord__register__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage)
            opts="-h -v -q --help --verbose --quiet generate guild user gate help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__gate)
            opts="-h -v -q --help --verbose --quiet list add remove help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__gate__add)
            opts="-h -v -q --help --verbose --quiet <GUILD_ID> <COLONY_ADDRESS> <DOMAIN_ID> <REPUTATION> <ROLE_ID>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__gate__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__gate__list)
            opts="-g -a -h -v -q --guild --all-guilds --help --verbose --quiet <START> <END>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --guild)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -g)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__gate__remove)
            opts="-h -v -q --help --verbose --quiet <GUILD_ID> <COLONY_ADDRESS> <DOMAIN_ID> <REPUTATION> <ROLE_ID>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__generate)
            opts="-h -v -q --help --verbose --quiet"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__guild)
            opts="-h -v -q --help --verbose --quiet list remove help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__guild__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__guild__list)
            opts="-h -v -q --help --verbose --quiet <START> <END>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__guild__remove)
            opts="-h -v -q --help --verbose --quiet <GUILD_ID>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__user)
            opts="-h -v -q --help --verbose --quiet list add remove help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__user__add)
            opts="-h -v -q --help --verbose --quiet <USER_ID> <WALLET_ADDRESS>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__user__help)
            opts="-v -q --verbose --quiet <SUBCOMMAND>..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__user__list)
            opts="-h -v -q --help --verbose --quiet <START> <END>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        discord__gating__bot__storage__user__remove)
            opts="-h -v -q --help --verbose --quiet <USER_ID>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

complete -F _discord-gating-bot -o bashdefault -o default discord-gating-bot
